pkg/controller/knicluster/knicluster_controller.go

```diff
@@ -251,6 +279,28 @@ func (r *ReconcileKNICluster) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Add conditions if there are none
+	if instance.Status.Conditions == nil {
+		conditions.SetStatusCondition(&instance.Status.Conditions, conditions.Condition{
+			Type:   conditions.ConditionAvailable,
+			Status: corev1.ConditionFalse,
+		})
+		conditions.SetStatusCondition(&instance.Status.Conditions, conditions.Condition{
+			Type:   conditions.ConditionProgressing,
+			Status: corev1.ConditionTrue,
+		})
+		conditions.SetStatusCondition(&instance.Status.Conditions, conditions.Condition{
+			Type:   conditions.ConditionDegraded,
+			Status: corev1.ConditionFalse,
+		})
+		conditions.SetStatusCondition(&instance.Status.Conditions, conditions.Condition{
+			Type:   conditions.ConditionUpgradeable,
+			Status: corev1.ConditionUnknown,
```

Comment:
Should this intentionally remain in "Unknown" state? I don't see anything else changing its value.

Reply:
I think I'm going to remove it for now.

Reply:
Decided to mark it as upgradeable when reconcile completes successfully